spec/index.adoc

```diff
@@ -1674,6 +1674,96 @@ Any user can top-up any canister this way.
+
+Information about Bitcoin can be found in the https://developer.bitcoin.org/devguide/[Bitcoin developer guides].
+
+[#ic-bitcoin-candid]
+=== Interface overview
+
+The following interface description, in https://github.com/dfinity/candid/blob/master/spec/Candid.md[Candid syntax], describes the available functionality. You can also link:{attachmentsdir}/bitcoin.did[download the file].
+----
+include::{example}bitcoin.did[]
+----
+
+The binary encoding of arguments and results are as per Candid specification.
+
+[#ic-bitcoin_get_utxos]
+=== IC method `get_utxos`
+
+Given a https://en.bitcoin.it/wiki/Base58Check_encoding[base58-encoded] address as part of the
+`GetUtxosRequest` , the function returns all unspent transaction outputs (UTXOs) associated with the
+provided address based on the current view of the Bitcoin blockchain available to the Bitcoin canister.
+
+If the call fails, e.g., because the address is malformed, a `GetUtxosError` variant is returned,
+indicating the reason for the failed call.
+
+The optional `min_confirmations` parameter can be used to limit the returned UTXOs to those with at
+least the provided number of confirmations, i.e., transactions with fewer than this number of
+confirmations are not considered.
+If this parameter is not used, the default value is 0.
+
+The optional `offset` parameter can be used to specify a starting offset in the list of UTXOs.
+Since a response will contain at most `TBD` `Utxo` records, this parameter is useful for addresses
+with many UTXOs.
+There is no guarantee that the set of UTXOs will remain unchanged between function calls with different
+offsets, i.e., every call will return the UTXOs starting from the provided offset based on the
+current view.
+If this parameter is not used, the default value is 0.
+
+[#ic-bitcoin_get_balance]
+=== IC method `get_balance`
+
+Given a https://en.bitcoin.it/wiki/Base58Check_encoding[base58-encoded] address as part of the
+`GetBalanceRequest` , the function returns the current balance of this address in `Satoshi` (10^8 Satoshi = 1 Bitcoin).
+
+If the call fails, e.g., because the address is malformed, a `GetBalanceError` variant is returned,
+indicating the reason for the failed call.
+
+The optional `min_confirmations` parameter can be used to limit the set of considered UTXOs
+for the calculation of the balance to those with at least the provided number of confirmations.
+
+Given an address and the optional `min_confirmations` parameter, `get_balance` iterates over
+all UTXOs returned by `get_utxos` for the same two parameters and returns the sum of
+UTXO values.
+Since the function is executed on the replica, there is no pagination limit, i.e., `get_balance`
+will iterate over all relevant UTXOs.
+
+[#ic-send_transaction]
+=== IC method `send_transaction`
+
+Given a `blob` of a Bitcoin transaction, several checks are performed:
+
+- The transaction is well formed.
+- The transaction only consumes unspent outputs.
```

Comment:
With respect to which state do you check that (a) the UTXOs were created and (b) they are unspent?

Reply:
With respect to the state of the canister you're talking to (i.e. the mainnet state if talking to the btc mainnet canister, and the bitcoin testnet if talking to the btc testnet canister).

Reply:
I assume that means we use the longest branch known to the Bitcoin canister if there are multiple ones, and the behavior is undefined in case there are two equally long branches (where one may create/spend the UTXO and the other one does not)?